Document instance migration phases 3 and 4 (#4348)
camunda/camunda-docs: docs/apis-tools/zeebe-api/gateway-service.md

- Not all active elements in the given process instance are mapped to the elements in the target process definition
- A mapping instruction changes the type of an element or event
- A mapping instruction changes the implementation of a task

> - A mapping instruction detaches a boundary event from an active element

Author: ❌ We still need adjust this to cover all catch events
```suggestion
- A mapping instruction detaches a catch event from an active element
```

Reviewer: @korthout this limitation was outdated as I added an example after it.  I refactored it to catch events in general but keep the example for boundary event. Please take another look. 👀

- A mapping instruction refers to an unsupported element (i.e. some elements will be supported later on)
- A mapping instruction refers to element in unsupported scenarios.
  (i.e. migrating active elements with event subscriptions will be supported later on)

> - Multiple mapping instructions target the same catch event
> - A mapping instruction detaches a boundary event from an active element
> - Two catch events in the source are mapped to the same catch event in the target

Author: ❌ We can combine both 

>Two catch events in the source are mapped to the same catch event in the target

and 

>Multiple mapping instructions target the same catch event 

```suggestion
- A mapping instruction detaches a boundary event from an active element
- Multiple mapping instructions refer to the same catch event
```

> - A catch event in the source is mapped to a different type of catch event in the target

Author: ❌ This is already covered by 

>A mapping instruction changes the type of an element **or event**

> - A message catch event in the source process instance is not fully distributed

Reviewer: Ah yeah, message catch event subscription it is 👍

> - Parallel multi-instance body is mapped to a sequential multi-instance body and vice versa

Author: 🔧 Let's use the same style for these sentences:

```suggestion
- A mapping instruction changes a parallel multi-instance body to a sequential multi-instance body or vice versa
```
